pkg/backend/telemetryapi/request.go

```diff
@@ -33,17 +33,36 @@ type requestsBuilder interface {
 }
 
 var (
-	errUnableToSplit = fmt.Errorf("unable to split large payload further")
+	errUnableToSplit     = fmt.Errorf("unable to split large payload further")
+	maxEntitiesByRequest = 100
 )
 
 func newBatchRequest(ctx context.Context, metricsBatch []metricBatch, apiKey string, url string, userAgent string) (reqs []request, err error) {
-	// todo: split payload based on:
-	// a) number of entities being sent
-	// b) payload size
+	// todo: split payload based on payload size
 	if len(metricsBatch) < 1 {
 		return nil, nil
 	}
 
+	if len(metricsBatch) <= maxEntitiesByRequest {
+		return buildRequests(ctx, metricsBatch, apiKey, url, userAgent)
+	}
+
+	metrics := metricsBatch[:min(len(metricsBatch), maxEntitiesByRequest)]
+	req, err := buildRequests(ctx, metrics, apiKey, url, userAgent)
+	reqs = append(reqs, req...)
+
+	if len(metricsBatch[maxEntitiesByRequest:]) > 0 {
```

Comment:
shouldn't be `min(len(metricsBatch), maxEntitiesByRequest)`?

Reply:
Yes!! Good catch.

I wonder why my test didn't catch the error. Will see

Reply:
Wait, no. Actually I don't need min func at all since if I've receive less metric than the max it will be cover in [L#46](https://github.com/newrelic/infrastructure-agent/pull/201/files/eed0ee1afc0262d0585aa95824597182d1cd7d41#diff-7317221c4741444842199690b44f321c4c4c140708ead04cc8937a364516024cR46)